### PR TITLE
Add templated method SetCurlEasySetOpt

### PR DESCRIPTION
**Pull request by prestonfromspotx**

### CODE REVIEW

File: inc/lift/Request.h [hunk 1]
```diff
@@ -241,7 +241,18 @@ class Request {
      * @param verify the verify value to set the CURLOPT_SSL_VERIFYHOST option to
      */
     auto SetVerifySSLHost(int64_t verify) -> void;
-
+
```
Comment (tcashel) [inline inc/lift/Request.h:244]: **this is a test comment** 
`code text how fancy`

File: inc/lift/Request.h [hunk 2]
```diff
@@ -241,7 +241,18 @@ class Request {
      * @param verify the verify value to set the CURLOPT_SSL_VERIFYHOST option to
      */
     auto SetVerifySSLHost(int64_t verify) -> void;
-
+    
+    /**
+     * Method to allow users to set any CURLoption with any value type
+     * @tparam OptionValueType Type of the option value to set in curl_easy_setopt
+     * @param curl_option_name CURLoption enum indicating what CURL option will be changed
+     * @param option_value     The value to set the CULR option to
```
Comment (kevin-logan) [inline inc/lift/Request.h:249]: CULR != CURL

**Review by tcashel — CHANGES_REQUESTED** [review]

File: inc/lift/Request.h [hunk 3]
```diff
@@ -241,7 +241,18 @@ class Request {
...
+     */
+    template<typename OptionValueType>
+    inline auto SetCurlEasySetOpt(CURLoption curl_option_name, OptionValueType option_value) -> void {
```
Comment (kevin-logan) [inline inc/lift/Request.h:252]: take OptionValueType by && and forward it to curl for perfect forwarding

File: inc/lift/Request.h [hunk 4]
```diff
@@ -241,7 +241,18 @@ class Request {
      * @param verify the verify value to set the CURLOPT_SSL_VERIFYHOST option to
      */
     auto SetVerifySSLHost(int64_t verify) -> void;
-
+    
+    /**
+     * Method to allow users to set any CURLoption with any value type
```
Comment (kevin-logan) [inline inc/lift/Request.h:246]: I'd like a comment indicating that this should only be used if the feature isn't already a part of liblift, and ideally one should consider adding it to liblift before using this

**Review by kevin-logan — CHANGES_REQUESTED** [review]

**Pull request closed without merging**